### DISCUSSION TITLE
6.1.2 instability?

### DISCUSSION THREAD

**Discussion by jslitzkerttcu** [chuot/rdio-scanner#81] [Q&A]
Is anyone else experiencing instability with 6.1.2?  I have had good luck with 6.1 but after updating to 6.1.2, the software is crashing and requiring me to stop the service and restart it, then it has to ingest all the calls that have backlogged since it went down.  Simply curious if this is an isolated issue I'm experiencing or if anyone else has the same issue.

**Comment by noamlivne** [comment]
Probably happened only once to me a few hours after the first installation. Since then a few days have passed + I stopped and started the service on purpose a few times, and did not encounter any issues.
I am using Windows 10 + SDRTRUNK with Tailscale as proxy.

**Reply by jslitzkerttcu** [reply]
Thanks!  I'll keep an eye on it.  This is the second time it's done it but we'll see.  Nothing in the logs so far when it crashed.  Running Linux Ubuntu.

**Reply by chuot** [reply]
If it crashes, the Rdio Scanner logs won't have any entries about it. Is there anything about the crash if look at the logs from `sudo journalctl -u rdio-scanner.service`?

**Reply by jslitzkerttcu** [reply]
Found it!  I wonder if it's just from trying to ingest all the files at once from being offline or something?  It's been online and up since yesterday with no issues.  Here's what I found in the journal (see attached).
[error.log](https://github.com/chuot/rdio-scanner/files/7836011/error.log)

**Reply by noamlivne** [reply]
After days of running excellently non-stop, the app stopped ingesting again. According to the time it happened it resembles to what happened about a week ago, but I cannot reproduce. 
Perhaps removing the Rdio-Scanner app (and Tailscale app) from Android's memory, causes the Windows program to stop ingesting.
I did it before A LOT of times with no issues. So it only happens rarely:

2022/01/13 14:49:43 client.conn.writemessage: write tcp 100.79.29.86:443->100.89.248.62:43434: i/o timeout

Anyway, I will try 6.1.3

**Reply by noamlivne** [reply]
The program stopped to ingest also with 6.1.3. There is no Error this time and the program continue to Prune, but not to ingest.
Perhaps it is related to swiping Tailscale and Rdio-scanner from Android's memory. Though I do that quite often, so I do not know the exact cause.
Windows 10 with SDRTrunk.
I'll learn to live with it.

**Comment by chuot** [comment]
It appears that the thread running the dirwatch is crashing without giving much information is regards to the crash. In fact, the [fsnotify](https://github.com/fsnotify/fsnotify) package, used by dirwatch, crashed for some reason without giving much informations in regards to the crash. Since I'm not testing on a regular basis if the dirwatch thread is still running, I will add some checks on v6.1.4 to restart the dirwatch thread if it stops.
